XMonad/Hooks/StatusBar.hs

```diff
@@ -311,9 +313,19 @@ statusBarPropTo prop cmd pp = def
 statusBarPipe :: String -- ^ The command line to launch the status bar
               -> X PP   -- ^ The pretty printing options
               -> IO StatusBarConfig
-statusBarPipe cmd xpp  = do
-    h <- spawnPipe cmd
-    return $ def { sbLogHook = xpp >>= \pp -> dynamicLogWithPP pp { ppOutput = hPutStrLn h } }
+statusBarPipe cmd xpp = do
+    hRef <- newIORef Nothing
+    return $ def
+        { sbStartupHook = io (writeIORef hRef . Just =<< spawnPipe cmd)
+        , sbLogHook     = do
+              pp <- xpp
+              h' <- io (readIORef hRef)
+              whenJust h' $ \h -> dynamicLogWithPP pp { ppOutput = hPutStrLn h }
+        , sbCleanupHook = io
+                          $   readIORef hRef
+                          >>= (`whenJust` hClose)
+                          >>  writeIORef hRef Nothing
```

Comment:
I mean, that's what `brittany` thought the output should look like!

Reply:
I can appreciate trying to align everything :P  But I agree, on second look, that this looks perhaps a bit weird

Reply:
In general, is there a formatter or some presets you use? I don't like manually formatting code

Reply:
I format imports with `fourmolu` (when appropriate), but other than that not really.  Most formatters kill alignment (or align in weird ways) and since I like pretty things I'm stuck with formatting things myself

Reply:
I also tried `fourmolu` a couple days ago and it mostly matches my style so I might use it in new code, but in an old codebase it ends up reformatting too much unrelated code. Not sure if there's a good solution here. We could possibly settle on a style and then reformat everything and enforce that formatting in CI, but it might be too much of a barrier to contributors. And if we don't do this in CI, then the code inevitably becomes hand-formatted again.

Reply:
I don't think we should enforce a style, at least in the contrib modules, since it should have a relatively low barrier to entry. People will probably work on different files anyway, it's just important to keep the style consistent within that file

Reply:
Turns out we have a different notion of consistency. :-)

I found it inconsistent that the `do` block is indented by 4 but the `io $ …` is indented to horizontally align, but it turns out that there are at least two other blocks in the file that are also horizontally aligned, so it could be said that some stuff is consistently aligned horizontally.

Anyway, if you like that alignment, feel free to drop https://github.com/xmonad/xmonad-contrib/pull/553/commits/3ee821c656f5e4b51f2d513e4234b214c10df601 and merge it as it was. Or merge it as it is. No need to bikeshed this further.

Reply:
I dropped that commit since I'm not sure what you mean and I don't think it'll make a huge difference

Reply:
Just so we all know what I mean, I mean this (in)consistency:

![image](https://user-images.githubusercontent.com/300342/120493249-14312d80-c3bb-11eb-923d-ea34109ac1cc.png)